target_dynamics_v2/target.py

```diff
@@ -77,15 +78,29 @@ def validate_fields_mapping(self, fields_mapping: dict):
             not_overridable_fields = override_fields_name - set(sink.allowed_fields_override)
             if not_overridable_fields:
                 raise InvalidCustomFieldDefinition(f"Non-overridable fields provided in config for sink={sink.name}, fields={not_overridable_fields}")
-    
+
+    def get_tenant_config(self):
+        snapshot_directory = self.config.get("snapshot_dir", None)
+        if snapshot_directory == None:
+            raise InvalidConfigurationError("snapshot_dir is not provided in the config")
+
+        config_path = os.path.join(snapshot_directory, "tenant-config.json")
+
+        if not os.path.exists(config_path):
+            raise InvalidConfigurationError(f"tenant-config.json does not exist in the snapshot directory={snapshot_directory}")
+
+        with open(config_path) as f:
+            tenant_config = json.load(f)
+
+        return tenant_config
+
+
     def load_fields_and_dimensions_mapping_config(self):
-        config_path = f"../vinni-tenant-config.json"
+        tenant_config = self.get_tenant_config()
+        config = tenant_config.get("dynamics_bc_field_mapping")
 
-        config= {}
-        # Verifies if `tenant-config.json` exists
-        if os.path.exists(config_path):
-            with open(config_path) as f:
-                config = json.load(f)
+        if config == None:
+            raise InvalidConfigurationError("dynamics_bc_field_mapping is not provided in the tenant-config.json")
```

Comment:
I`m not sure having the field mapping config is a requirement. We can check that with @izkapadia-hg